### DISCUSSION TITLE
The filtered field in a large-eddy simulation

### DISCUSSION THREAD

**Discussion by ChiyodaMomo01** [CliMA/Oceananigans.jl#2948] [General]
In a large-eddy simulation, a filter is used to divide a spatial and temporal field into two components: the filtered field and the sub-filtered field (denoted by a prime). Does `Oceananigans.jl` provide a means to output the filtered field at each timestep?

**Comment by simone-silvestri** [comment]
The filtering in Oceananigans' LES is _implicit_.
We are directly solving for the "filtered" field, while the effect of the "subfiltered" (or subgrid) velocity component on the filtered (resolved) field is accounted for with a _sub-grid-scale_ closure. 

I.e., we solve for

$\frac{\partial \hat{u}_i }{\partial t} + \frac{\partial  \hat{u}_j \hat{u}_i}{\partial x_j} = \frac{\partial \hat{p}}{\partial x_i} + \frac{\partial \tau}{\partial x_j}- \frac{\partial \widehat{u_j^\prime u_i^\prime}}{\partial x_j}$ 

where a hat implies an (implicit!) filter operation and $\tau$ is the filtered viscous stress defined as 

$\nu \left(\frac{\partial \hat{u}_j}{\partial x_i} + \frac{\partial \hat{u}_i}{\partial x_j}\right)$.

The output of our model are the "filtered" fields $\hat{u}, \hat{v}, \hat{w}, \hat{p}$. 
Take care that the filter operation is not actually performed but it is implied in the equations we are solving.
With this formulation, the rightmost term which represents the mixing of the resolved flow due to subgrid-scale velocities, is unclosed. 
We usually treat this term with a turbulent viscosity approximation where
$\nu_e \left(\frac{\partial \hat{u}_j}{\partial x_i} + \frac{\partial \hat{u}_i}{\partial x_j}\right) \approx \widehat{u_j^\prime u_i^\prime}$.

We have several models to calculate $\nu_e$ in Oceananigans (Leith, Smagorinsky, Anisotropic Minimum Dissipation...). 
With Oceananigans you cannot explicitly retrieve the subgrid-scale velocities, only the turbulent fluxes.

**Reply by tomchor** [reply]
Great explanation @simone-silvestri. I'd just like to add that it's not just Oceananigans that is like this. The great majority of LES models uses the same approach (i.e. implicit and not explicit filtering). In fact I've never personally encountered a LES model that uses an explicit filter (although I'm sure some must exist).

**Reply by glwagner** [reply]
There are lots of papers on explicit filtering for LES, but I agree its rare in practice due to the additional expense. Here's one of the more prominent papers on explicit filtering:

https://aip.scitation.org/doi/full/10.1063/1.3485774?casa_token=U73-ObYd_rQAAAAA%3AWffgzgXHKOfcF0ZDvsK5moesijtn5SqtAaLI5qDY108zCex9ucrwa2Xm-icRbyhwkGUpBHDhbDw

**Comment by glwagner** [comment]
> Does Oceananigans.jl provide a means to output the filtered field at each timestep?

To add to @simone-silvestri's great response, I want to point out that it certainly is possible to implement filtering using `Callback`. If one goes this route, the outputted fields would be filtered. `Oceananigans` is designed to be flexible, so --- generally speaking --- it is possible to output any information computed during a simulation.
